cfme/tests/cloud/test_instance_power_control.py

```diff
@@ -44,16 +44,16 @@ def vm_name(testing_instance):
     return testing_instance.name
 
 
-def wait_for_state_change_time_refresh(instance, state_change_time, timeout=300):
+def wait_for_state_change_time_refresh(instance, provider, state_change_time, timeout=300):
     """ Waits for 'State Changed On' refresh
     """
     def _wait_for_state_refresh():
         instance.load_details()
         return state_change_time != instance.get_detail(
             properties=("Power Management", "State Changed On"))
-
     try:
-        wait_for(_wait_for_state_refresh, num_sec=timeout, delay=30)
+        wait_for(_wait_for_state_refresh, fail_func=provider.refresh_provider_relationships(),
```

Comment:
I think this is too much refreshing, you should look at the ```RefreshTimer``` usage, there are some good examples in the code base. This allows you to test every 30 seconds, but to only initiate the ```refresh_provider_relationships``` maybe every3 minutes.